src/Mailer/PreviewResult.php

```diff
@@ -55,7 +56,9 @@ protected function processMailer(Mailer $mailer, $method)
             ->getMethod('_renderTemplates')
             ->getClosure($email);
 
+        Router::setRequestInfo(['plugin' => null]);
```

Comment:
Sorry if I am misunderstanding, but I don't think this patch fixes the issue. Even after your fix, it wouldn't work correctly if you want to preview emails from plugins or prefixed actions,. For example, if you send an email in an admin action, a URL array `['controller' => 'Pages', 'action' => 'index']` must be `/admin/pages`, but it will be `/pages` in the mail preview. It is not correct. We would have to add a feature that allows people to tell the original request context to DebugKit, instead of this kind of hack.